sgt-back-end/index.js

```diff
@@ -0,0 +1,126 @@
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const express = require('express');
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  SELECT *
+  FROM "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      const allGrades = result.rows;
+      res.status(200).json(allGrades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  if (!name || !course || score === undefined) {
+    res.status(400).json({ error: 'Invalid or missing. name, course and score are required' });
+  } else if (!Number.isInteger(score) || score < 0 || score > 100) {
+    res.status(400).json({ error: 'Please provide a valid score between 0-100' });
+    return;
+  }
+  const params = [name, course, score];
+  const sql = `
+  INSERT into "grades" ("name", "course", "score")
+  VALUES ($1, $2, $3)
+  RETURNING *
+  `;
+
+  db.query(sql, params)
+    .then(result => {
+      const newGrade = result.rows[0];
+      res.status(201).json(newGrade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({ error: '"gradeId" must be a positive integer' });
+  } else if (!name || !course || score === undefined) {
+    res.status(400).json({ error: 'Invalid or missing. name, course and score are required.' });
+  } else if (score < 0 || score > 100) {
+    res.status(400).json({ error: 'Please provide a score between 0-100' });
+  }
```

Comment:
Scores also need to be an integer

Reply:
@francismanalang It looks like you addressed the POST, now please address the PUT